### PR TITLE
DTWRKS-1110: Reingest main snapshot with optimisations

### PR DESCRIPTION
**Pull request by nathanbigaignon**

### CODE REVIEW

File: steps/ingesters.py [hunk 1]
```diff
@@ -257,97 +257,36 @@ def __init__(self, configuration, collection_name, spark_session, hive_session):
...
-                                                        prefix="corporate_data_ingestion/hive/external/dwx_audit_transition.db/calc_parts_snapshot_enriched_unpartitioned/")
-
-        df = self._spark_session.read.orc(s3_source_url)
-        (
-            df.coalesce(1000)
             .withColumn("id_part", df.id_key[0:2])
             .select("id_key", "json", "dbType", "id_part")
-            .write.insertInto("dwx_audit_transition.calc_parts_snapshot", overwrite=True)
+            .repartition("id_part", "dbType")
+            .sortWithinPartitions("id_part", "dbType")
```
Comment (nicholasjgreen) [inline steps/ingesters.py]: don't we want to sort by id_key?

File: steps/ingesters.py [hunk 2]
```diff
@@ -257,97 +257,36 @@ def __init__(self, configuration, collection_name, spark_session, hive_session):
...
+            .bucketBy(256, 'id_key')
+            .write.insertInto("dwx_audit_transition.calculation_parts_snapshot", overwrite=True)
```
Comment (nicholasjgreen) [inline steps/ingesters.py:289]: do we want to overwrite this table or put it somewhere else?

File: locals.tf [hunk 3]
```diff
@@ -245,7 +245,7 @@ locals {
     qa          = 150
     integration = 150
     preprod     = 170
-    production  = 170
+    production  = 269
```
Comment (nicholasjgreen) [inline locals.tf:248]: did we do the maths to make sure this many will fit?

**Review by nicholasjgreen — CHANGES_REQUESTED** [review]

**Review by nicholasjgreen — APPROVED** [review]

File: locals.tf (hunk 3 above)
Comment (nathanbigaignon) [reply locals.tf:248]: yep (number of executors per instance * number of core instances) minus 1 for the driver

**Pull request merged**